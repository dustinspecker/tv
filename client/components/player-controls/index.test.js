import IconButton from '@material-ui/core/IconButton'
import PlayerControls from '.'
import React from 'react'
import renderer from 'react-test-renderer'
import Slider from '@material-ui/lab/Slider'

test('PlayerControls hides controls when showControls false', () => {
  const props = {
    showControls: false
  }

  const element = <PlayerControls {...props} />

  const component = renderer.create(element)

  expect(component.toJSON()).toMatchSnapshot()
})

test('PlayerControls Play/Pause button calls playOrPause', () => {
  const props = {
    currentTime: 0,
    playOrPause: jest.fn(),
    showControls: true
  }

  const element = <PlayerControls {...props} />

  const component = renderer.create(element)

  const playButton = component.root.findAllByType(IconButton)[0]

  playButton.props.onClick()

  expect(props.playOrPause.mock.calls.length).toEqual(1)
})

test('PlayerControls Slider onChange call seek with value', () => {
  const props = {
    currentTime: 0,
    seek: jest.fn(),
    showControls: true
  }

  const element = <PlayerControls {...props} />

  const component = renderer.create(element)

  const slider = component.root.findByType(Slider)

  slider.props.onChange(null, 73)

  expect(props.seek.mock.calls.length).toEqual(1)
  expect(props.seek.mock.calls[0][0]).toEqual(73)
})

test('PlayerControls Stop button calls stop', () => {
  const props = {
    currentTime: 0,
    showControls: true,
    stop: jest.fn()
  }

  const element = <PlayerControls {...props} />

  const component = renderer.create(element)

  const stopButton = component.root.findAllByType(IconButton)[1]

  stopButton.props.onClick()

  expect(props.stop.mock.calls.length).toEqual(1)
})

test('PlayerControl shows human readable time', () => {
  const props = {
    currentTime: 63,
    duration: 659,
    showControls: true
  }

  const element = <PlayerControls {...props} />

  const component = renderer.create(element)

  expect(() => {
    component.root.findByProps({children: '1:03'})
    component.root.findByProps({children: '10:59'})
  }).not.toThrow()
})
