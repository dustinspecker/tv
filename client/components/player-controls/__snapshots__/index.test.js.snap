// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerControl shows human readable time 1`] = `
<header
  className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-4 MuiAppBar-positionFixed-5 MuiAppBar-colorDefault-10 mui-fixed PlayerControls-appBar-1"
>
  <div
    className="MuiToolbar-root-40 MuiToolbar-regular-42 MuiToolbar-gutters-41"
  >
    <button
      className="MuiButtonBase-root-50 MuiIconButton-root-44"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-49"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-53"
        >
          pause
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-76"
      />
    </button>
    <div>
      1:03
    </div>
    <div
      aria-orientation="horizontal"
      aria-valuemax={659}
      aria-valuemin={0}
      aria-valuenow={63}
      className="MuiSlider-root-62 PlayerControls-sliderRoot-2"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchMove={[Function]}
      onTouchStartCapture={[Function]}
      role="slider"
      variant="determinate"
    >
      <div
        className="MuiSlider-container-63"
      >
        <div
          className="MuiSlider-track-64 PlayerControls-sliderTrack-3 MuiSlider-trackBefore-65"
          style={
            Object {
              "transform": "translateY(-50%) scaleX(0.09559939301972685)",
            }
          }
        />
        <div
          className="MuiSlider-thumbWrapper-67"
          style={
            Object {
              "transform": "translateX(9.559939301972685%)",
            }
          }
        >
          <button
            className="MuiButtonBase-root-50 MuiSlider-thumb-68"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            onTouchStartCapture={[Function]}
            tabIndex="0"
            type="button"
          />
        </div>
        <div
          className="MuiSlider-track-64 PlayerControls-sliderTrack-3 MuiSlider-trackAfter-66"
          style={
            Object {
              "transform": "translateY(-50%) scaleX(0.9044006069802731)",
            }
          }
        />
      </div>
    </div>
    <div>
      10:59
    </div>
    <button
      className="MuiButtonBase-root-50 MuiIconButton-root-44"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-49"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-53"
        >
          stop
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-76"
      />
    </button>
  </div>
</header>
`;

exports[`PlayerControls hides controls when showControls false 1`] = `null`;
